source/Octopus.Tentacle.Tests.Integration/Util/Builders/Decorators/Proxies/MethodLoggingProxyDecorator.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Serilog;
+
+namespace Octopus.Tentacle.Tests.Integration.Util.Builders.Decorators.Proxies
+{
+    public class MethodLoggingProxyDecorator : ServiceProxy
+    {
+        readonly ILogger logger = new SerilogLoggerBuilder().Build().ForContext<MethodLoggingProxyDecorator>();
+        string serviceTypeName;
+
+        void Configure(string serviceTypeName)
+        {
+            this.serviceTypeName = serviceTypeName;
+        }
+
+        public static TService Create<TService>(TService service) where TService : class
+        {
+            var proxiedService = DispatchProxyAsync.Create<TService, MethodLoggingProxyDecorator>();
+            var proxy = proxiedService as MethodLoggingProxyDecorator;
+            proxy!.SetTargetService(service);
+            proxy.Configure(typeof(TService).Name);
+
+            return proxiedService;
+        }
+
+        protected override void OnStartingInvocation(MethodInfo targetMethod)
+        {
+            logger.Information("{ServiceName:l}.{MethodName:l}() started", serviceTypeName, targetMethod.Name);
+        }
+
+        protected override Task OnStartingInvocationAsync(MethodInfo targetMethod)
```

Comment:
Is there a reason this isn't an async method? I noticed you declare the async methods in MethodInvocationHookProxyDecorator using the async keyword.

Reply:
Yeah, my personal preference is to not decorate a method with the `async` keyword if I'm not using an `await` inside the method. In those cases, I just return a completed task.

This is just a personal preference, so if this is something you'd like changed to `async Task`, then I can do that :)